Build Error in ADIOS 2.5.0 on OSX: JsonWorker

Reference: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=76704 in https://github.com/conda-forge/adios2-feedstock/pull/8

Compiler: Clang 4.0.1 on Apple-Darwin13.4.0

CMake feature matrix:
```
ADIOS2 build configuration:
  ADIOS Version: 2.5.0
  C++ Compiler : Clang 4.0.1 
    /usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/x86_64-apple-darwin13.4.0-clang++

  Installation prefix: /usr/local/miniconda/conda-bld/adios2_1569893413423/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh
        bin: bin
        lib: lib
    include: include
      cmake: lib/cmake/adios2
     python: lib/python3.6/site-packages

  Features:
    Library Type: shared
    Build Type:   Release
    Testing: OFF
    Examples: OFF
    Build Options:
      Blosc    : ON
      BZip2    : OFF
      ZFP      : OFF
      SZ       : OFF
      MGARD    : OFF
      PNG      : ON
      MPI      : ON
      DataMan  : ON
      Table    : ON
      SSC      : ON
      SST      : ON
      DataSpaces: OFF
      ZeroMQ   : ON
      HDF5     : ON
      Python   : ON
      Fortran  : OFF
      SysVShMem: ON
      Profiling: ON
      Endian_Reverse: OFF
    RDMA Transport for Staging: Unconfigured
```

Compile error:
```
/usr/local/miniconda/conda-bld/adios2_1569893413423/work/source/adios2/toolkit/query/JsonWorker.cpp:128:37: error: call to member function 'compare' is ambiguous
    if (m_SourceReader->m_IO.m_Name.compare(ioName) != 0)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1189:9: note: candidate function
    int compare(const basic_string& __str) const _NOEXCEPT;
        ^
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1191:9: note: candidate function
    int compare(__self_view __sv) const _NOEXCEPT;
        ^
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1205:9: note: candidate function not viable: no known conversion from 'nlohmann::basic_json<std::map, std::vector, std::__1::basic_string<char>, bool, long long, unsigned long long, double, std::allocator, adl_serializer>' to 'const value_type *' (aka 'const char *') for 1st argument
    int compare(const value_type* __s) const _NOEXCEPT;
        ^
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1204:9: note: candidate function template not viable: requires at least 4 arguments, but 1 was provided
        compare(size_type __pos1, size_type __n1, const _Tp& __t, size_type __pos2, size_type __n2=npos) const;
        ^
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1193:9: note: candidate function not viable: requires 3 arguments, but 1 was provided
    int compare(size_type __pos1, size_type __n1, __self_view __sv) const;
        ^
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1195:9: note: candidate function not viable: requires 3 arguments, but 1 was provided
    int compare(size_type __pos1, size_type __n1, const basic_string& __str) const;
        ^
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1206:9: note: candidate function not viable: requires 3 arguments, but 1 was provided
    int compare(size_type __pos1, size_type __n1, const value_type* __s) const;
        ^
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1207:9: note: candidate function not viable: requires 4 arguments, but 1 was provided
    int compare(size_type __pos1, size_type __n1, const value_type* __s, size_type __n2) const;
        ^
/usr/local/miniconda/conda-bld/adios2_1569893413423/_build_env/bin/../include/c++/v1/string:1196:9: note: candidate function not viable: requires at least 4 arguments, but 1 was provided
    int compare(size_type __pos1, size_type __n1, const basic_string& __str, size_type __pos2, size_type __n2=npos) const;
        ^
1 error generated.
```

--- a/source/adios2/toolkit/query/JsonWorker.cpp
+++ b/source/adios2/toolkit/query/JsonWorker.cpp
@@ -124,7 +124,7 @@ void JsonWorker::ParseJson()
     }
 
     auto ioO = jsonObj.find("io");
-    auto ioName = (*ioO)["name"];
+    std::string const ioName = (*ioO)["name"];
     if (m_SourceReader->m_IO.m_Name.compare(ioName) != 0)
         throw std::ios_base::failure("invalid query io. Expecting io name = " +
                                      m_SourceReader->m_IO.m_Name);

